notebook/static/notebook/js/notebook.js

```diff
@@ -2721,7 +2721,10 @@ define([
             return this.contents.get(this.notebook_path, {content: false}).then(
                 function (data) {
                     var last_modified = new Date(data.last_modified);
-                    if (last_modified > that.last_modified) {
+                    // We want to check last_modified (disk) > that.last_modified (our last save)
+                    // In some cases the filesystem reports an inconsistent time,
+                    // so we allow 0.5 seconds difference before complaining.
+                    if ((last_modified.getTime() - that.last_modified.getTime()) > 500) {  // 500 ms
```

Comment:
I don't think we need it to go in either direction - we want to show the message if the on-disk timestamp is newer than the timestamp from when we last saved it.  The other way round would be weird.